adapter/src/dummy.rs

```diff
@@ -190,43 +242,105 @@ impl Unlockable for Dummy {
     }
 }
 
+/// The dummy Header token used for the `Bearer` `Authorization` header
+///
+/// The format for the header token is:
+/// `{Auth token}:chain_id:{Chain Id}`
+#[derive(Debug, Clone, Display, FromStr)]
+#[display("{token}:chain_id:{chain_id}")]
+pub struct HeaderToken {
+    /// the Authentication Token
+    pub token: String,
+    /// The [`ChainId`] for which we authenticate
+    pub chain_id: ChainId,
+}
+
+mod deposit {
+    use crate::primitives::Deposit;
+    use dashmap::DashMap;
+    use primitives::{Address, ChainId, ChainOf, Channel, ChannelId};
+    use std::sync::Arc;
+
+    #[derive(Debug, Clone, Copy, Hash, Eq, PartialEq)]
+    pub struct Key {
+        channel_id: ChannelId,
+        chain_id: ChainId,
+        depositor: Address,
+    }
+
+    impl Key {
+        pub fn from_chain_of(channel_context: &ChainOf<Channel>, depositor: Address) -> Self {
+            Self {
+                channel_id: channel_context.context.id(),
+                chain_id: channel_context.chain.chain_id,
+                depositor,
+            }
+        }
+    }
+
+    /// Mocked deposits for the Dummy adapter.
+    ///
+    /// These deposits can be set once and the adapter will return
+    /// the set deposit on every call to [`get_deposit()`](crate::client::Locked::get_deposit).
+    #[derive(Debug, Clone, Default)]
+    pub struct Deposits(pub Arc<DashMap<Key, Deposit>>);
+
+    impl Deposits {
+        pub fn new() -> Self {
+            Self::default()
+        }
+
+        /// Get's the set deposit for the give [`ChannelId`] and [`Address`].
```

Comment:
a minor typo, should be

```suggestion
        /// Get's the set deposit for the given [`ChannelId`] and [`Address`].
```

Reply:
fixed + changed to Key (the actual used key for the deposits)